WordPress/src/main/java/org/wordpress/android/ui/domains/usecases/FetchAllDomainsUseCase.kt

```diff
@@ -0,0 +1,30 @@
+package org.wordpress.android.ui.domains.usecases
+
+import org.wordpress.android.fluxc.network.rest.wpcom.site.AllDomainsDomain
+import org.wordpress.android.fluxc.store.SiteStore
+import javax.inject.Inject
+
+class FetchAllDomainsUseCase  @Inject constructor(
+    private val siteStore: SiteStore,
+) {
+    suspend fun execute(): AllDomains {
+        val result = siteStore.fetchAllDomains()
+        return when {
+            result.isError -> AllDomains.Error
+            result.domains.isNullOrEmpty() -> return AllDomains.Empty
+            else -> result.domains?.run {
+                AllDomains.Success(this)
+            } ?: AllDomains.Empty
```

Comment:
Since there is the `isNullOrEmpty` check above, we should be safe here to use
```
else -> AllDomains.Success(result.domains!! /or/ requreNotNull(result.domains))
```

I just played with the code to make it a bit flattened just for fun and figured out something like:

```
suspend fun execute(): AllDomains {
    val result = siteStore.fetchAllDomains()
    if (result.isError) return AllDomains.Error
    val domains = result.domains.orEmpty().ifEmpty { return AllDomains.Empty }
    return AllDomains.Success(domains)
}
```

Don't take it as a strong recommendation :)

Reply:
Thanks again for the feedback :)
As you can imagine I chose the most complicated way to avoid the Not-Null Assertion Operator 😅 
I went with your 1st suggestion in https://github.com/wordpress-mobile/WordPress-Android/pull/19401